core/src/wallet/bitcoin/transaction_builders/BitcoinLikeStrategyUtxoPicker.cpp

```diff
@@ -473,11 +500,42 @@ namespace ledger {
             std::vector<BitcoinLikeUtxo> out;
 
             // Random shuffle utxos
-            std::vector<size_t> indexes(utxos.size());
-            std::iota(indexes.begin(), indexes.end(), 0);
+            const auto& sz = utxos.size();
+            std::vector<size_t> indexes;
+            indexes.reserve(sz);
+            {
+                auto const seed = std::chrono::system_clock::now().time_since_epoch().count();
+
+                if(useConfirmedFirst) {
 
-            auto const seed = std::chrono::system_clock::now().time_since_epoch().count();
-            std::shuffle(indexes.begin(), indexes.end(), std::default_random_engine(seed));
+                    std::vector<size_t> indexesTmp(sz, 0);
+                    std::iota(indexesTmp.begin(), indexesTmp.end(), 0);
+                    std::shuffle(indexesTmp.begin(), indexesTmp.end(), std::default_random_engine(seed));
+
+                    std::list<size_t> allindexes;
+                    std::list<size_t> confirmed;
+                    std::list<size_t> unconfirmed;
+
+                    for(const auto idx: indexesTmp) {
+
+                        if(utxos[idx].blockHeight.hasValue()) {
+                            confirmed.emplace_back(idx);
+                        } else {
+                            unconfirmed.emplace_back(idx);
+                        }
+                    }
+
+                    allindexes.splice(allindexes.end(), confirmed);
+                    allindexes.splice(allindexes.end(), unconfirmed);
+                    std::move(std::begin(allindexes), std::end(allindexes), std::back_inserter(indexes));
```

Comment:
Could be done other way: shuffle indexes & stable sort on confirmed/unconfirmed

Reply:
fixed in 15cc192b0c9fbb0e9ac50616bc4fac3252194e0c